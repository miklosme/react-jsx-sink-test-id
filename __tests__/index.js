const babel = require('babel-core');
const plugin = require('../src/index');

function transform(code) {
  return babel.transform(code,  {
    plugins: ['syntax-jsx', plugin],
  }).code;
}

describe('react-jsx-sink-test-id', () => {
  it('should wrap jsx to the sinker', () => {
    const source = `
      <Foo data-test="bar" />
    `;

    expect(transform(source)).toMatchSnapshot();
  });
  it('should not wrap jsx without testid', () => {
    const source = `
      <Foo color="yellow" />
    `;

    expect(transform(source)).toMatchSnapshot();
  });
  it('should work with attribute expressions', () => {
    const source = `
      import React from 'react';
      
      class App extends React.Component {
        render() {
          const a = <Foo color="red" data-test="hello" />
          const b = <Foo data-test="hi"><h1>text</h1></Foo>
          const c = <Foo color="red" data-test={10} />
          const d = <Foo color="red" data-test={<hr />} />
          const e = <Foo color="red" data-test={'hello' + 10} />
          const f = <Foo color="red" data-test={\`asd${20}\`} />
          return <Foo color="red" data-test={true ? 'true' : 'false'} />
        }
      }
    `;
    expect(transform(source)).toMatchSnapshot();
  });
  it('should wrap properly when nested', () => {
    const source = `
      import React from 'react';

      const x = <Foo color="red">
        <Bar data-test="hello" /> 
      </Foo>;
      
      <Bar data-test="hello" />;
      
      const y = <Bar data-test="hello" />;
      
      const z = <Baz render={<Foo data-test="foo" />} />;
      
      const e = <Foo>{x => <Foo value={x} data-test="foo" />}</Foo>
    `;
    expect(transform(source)).toMatchSnapshot();
  });
});