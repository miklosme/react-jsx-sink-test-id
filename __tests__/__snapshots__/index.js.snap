// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-jsx-sink-test-id should not wrap jsx without testid 1`] = `
"
<Foo color=\\"yellow\\" />;"
`;

exports[`react-jsx-sink-test-id should work with attribute expressions 1`] = `
"import _sinker from \\"babel-plugin-transform-react-jsx-sink-test-id/sinker\\";

import React from 'react';

class App extends React.Component {
  render() {
    const a = _sinker(<Foo color=\\"red\\" />, \\"hello\\");
    const b = _sinker(<Foo><h1>text</h1></Foo>, \\"hi\\");
    const c = _sinker(<Foo color=\\"red\\" />, 10);
    const d = _sinker(<Foo color=\\"red\\" />, <hr />);
    const e = _sinker(<Foo color=\\"red\\" />, 'hello' + 10);
    const f = _sinker(<Foo color=\\"red\\" />, \`asd20\`);
    return _sinker(<Foo color=\\"red\\" />, true ? 'true' : 'false');
  }
}"
`;

exports[`react-jsx-sink-test-id should wrap jsx to the sinker 1`] = `
"import _sinker from \\"babel-plugin-transform-react-jsx-sink-test-id/sinker\\";

_sinker(<Foo />, \\"bar\\");"
`;

exports[`react-jsx-sink-test-id should wrap properly when nested 1`] = `
"import _sinker from \\"babel-plugin-transform-react-jsx-sink-test-id/sinker\\";

import React from 'react';

const x = <Foo color=\\"red\\">
        {_sinker(<Bar />, \\"hello\\")} 
      </Foo>;

_sinker(<Bar />, \\"hello\\");

const y = _sinker(<Bar />, \\"hello\\");

const z = <Baz render={_sinker(<Foo />, \\"foo\\")} />;"
`;
